test/app/lib/cinematic/cinematicParser.spec.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-env jasmine */
+import { parseShot } from '../../../../app/views/play/cinematic/Command/CinematicParser'
+
+describe('parseShot', () => {
+  it('handles an empty shot', () => {
+    expect(parseShot({})).toEqual([[]])
+  })
+
+  it('parses setup correctly', () => {
+    const systems = [{
+      parseSetupShot: jasmine.createSpy().and.returnValue(['setup commands'])
+    }]
+    const shot = {
+      setupShot: 'Example setup shot'
+    }
+    const results = parseShot(shot, systems)
+    expect(systems[0].parseSetupShot).toHaveBeenCalledWith({ setupShot: 'Example setup shot' })
+    expect(results).toEqual([['setup commands']])
+  })
+
+  it('parses setup correctly with many systems', () => {
+    const systems = [{
+      parseSetupShot: jasmine.createSpy().and.returnValue(['setup command1'])
+    }, {
+      parseSetupShot: jasmine.createSpy().and.returnValue(['setup command2'])
+    }]
+    const shot = {
+      setupShot: 'Example setup shot'
+    }
+    const results = parseShot(shot, systems)
+    expect(systems[0].parseSetupShot).toHaveBeenCalledWith({ setupShot: 'Example setup shot' })
+    expect(systems[1].parseSetupShot).toHaveBeenCalledWith({ setupShot: 'Example setup shot' })
+    expect(results).toEqual([['setup command1', 'setup command2']])
+  })
+
+  it('parses nothing correctly', () => {
+    const systems = [{
+      parseDialogNode: jasmine.createSpy().and.returnValue(['dialog commands'])
+    }]
+    const shot = {
+      dialogNodes: ['DialogNode1', 'DialogNode2']
+    }
+    const results = parseShot(shot, systems)
+    expect(systems[0].parseDialogNode.calls.all().map(o => o.args)).toEqual([['DialogNode1'], ['DialogNode2']])
+    // I don't like this test but I couldn't figure out another way.
+    expect(JSON.stringify(results)).toEqual('[[{"commands":["dialog commands"]}],[{"commands":["dialog commands"]}]]')
```

Comment:
I think this is fine, otherwise we could alo match individual entries in the array returned. But this also looks fine.

Reply:
The stringified version works because the `new Command` that is wrapped around the  dialog node gets turned into the `"commands"` property in that object literal. I wonder if all JavaScript classes turn into an object literal when stringified?
Do you know how we could check equality between two  command objects or: Command == Command in Js?

Reply:
Not sure, but I wonder if we can check property wise, like `results.commands` to be equal to `expectedResult.commands`?